protodata/src/main/kotlin/io/spine/kanban/codegen/Expression.kt

```diff
@@ -0,0 +1,361 @@
+
+    /**
+     * Obtains a [FieldAccess] to the [field] of this message.
+     */
+    fun field(field: Field): FieldAccess =
+        FieldAccess(this, field.name, field.cardinalityCase)
+}
+
+/**
+ * A selector for an access method for a Protobuf message field.
+ *
+ * Depending on the field type, may allow to generate getters and setters for the field.
+ */
+class FieldAccess
+internal constructor(
+    private val message: Expression,
+    private val name: FieldName,
+    private val cardinality: CardinalityCase = SINGLE
+) {
+
+    /**
+     * A getter expression for the associated field.
+     */
+    val getter: MethodCall
+        get() {
+            val simpleAccess = MethodCall(message, getterName)
+            return when (cardinality) {
+                LIST -> immutableListClass.call("copyOf", listOf(simpleAccess))
+                MAP -> immutableMapClass.call("copyOf", listOf(simpleAccess))
+                else -> simpleAccess
+            }
+        }
+
+    /**
+     * Constructs a setter expression for the associated field.
+     */
+    fun setter(value: Expression): MethodCall =
+        MethodCall(message, setterName, listOf(value))
+
+    /**
+     * Constructs an `addField(..)` expression for the associated field.
+     */
+    fun add(value: Expression): MethodCall =
+        MethodCall(message, addName, listOf(value))
+
+    /**
+     * Constructs an `addAllField(..)` expression for the associated field.
+     */
+    fun addAll(value: Expression): MethodCall =
+        MethodCall(message, addAllName, listOf(value))
+
+    /**
+     * Constructs an `putField(..)` expression for the associated field.
+     */
+    fun put(key: Expression, value: Expression): MethodCall =
+        MethodCall(message, putName, listOf(key, value))
+
+    private val getterName: String
+        get() = when (cardinality) {
+            LIST -> getListName
+            MAP -> getMapName
+            else -> prefixed("get")
+        }
+
+    private val getListName: String
+        get() = "get${name.value.camelCase()}List"
+
+    private val getMapName: String
+        get() = "get${name.value.camelCase()}Map"
+
+    private val setterName: String
+        get() = when (cardinality) {
+            LIST -> addAllName
+            MAP -> putAllName
+            else -> prefixed("set")
+        }
+
+    private val addName: String
+        get() = prefixed("add")
+
+    private val addAllName: String
+        get() = prefixed("addAll")
+
+    private val putName: String
+        get() = prefixed("put")
+
+    private val putAllName: String
+        get() = prefixed("putAll")
+
+    private fun prefixed(prefix: String) =
+        "$prefix${name.value.camelCase()}"
+
+    override fun toString(): String {
+        return "FieldAccess[$message#${name.value}]"
+    }
+}
+
+/**
+ * An expression of a Java method call.
+ *
+ * Might be a static or an instance method. In the case of the former, the receiver expression
```

Comment:
I think it is "May be...".

Reply:
Changed to "Can be...". The sentence describes the physical ability, not a permission.